src/components/pages/Chat/RoomDropdown/index.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useRef } from 'react'
+import { useState } from 'react'
+import styled from 'styled-components'
+import {
+    AddMemberIcon,
+    CopyIcon,
+    DeleteIcon,
+    DownloadIcon,
+    EditMembersIcon,
+    SearchIcon,
+} from '../../../../icons'
+import { KARELIA } from '../../../../utils/css'
+
+type Props = {
+    button?: React.ReactElement
+}
+
+const DropDownContainer = styled.div`
+    margin-left: 10px;
+`
+const DropDownHeader = styled.div`
+    margin-bottom: 0.8em;
+    padding: 0 2em 0 1em;
+    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.15);
+    font-weight: 500;
+    font-size: 1.3rem;
+    color: #3faffa;
+    background: #ffffff;
+`
+const DropDownListContainer = styled.div`
+    position: absolute;
+    top: 75px;
+    right: 35px;
+    z-index: 100;
+`
+const DropDownList = styled.div`
+    width: 250px;
+    padding: 0;
+    margin: 0;
+    background: #ffffff;
+    border-radius: 10px;
+    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.1);
+    color: #59799c;
+    font-family: ${KARELIA};
+    font-size: 14px;
+    list-style-type: none;
+    border-spacing: 0px;
+
+    hr {
+        border: 0;
+        border-top: 1px solid #dee6ee;
+        padding: 0px;
+        margin: 0px;
+    }
+`
+const ListItem = styled.div`
+    padding: 10px 15px;
+    background-clip: padding-box;
+    align-items: center;
+    display: flex;
+
+    td {
+        line-height: 13px;
+        &:last-child {
+            margin-left: 15px;
+        }
+    }
+
+    :first-child {
+        padding-top: 15px;
+        border-radius: 10px 10px 0 0;
+    }
+
+    :last-child {
+        padding-bottom: 15px;
+        border-radius: 0 0 10px 10px;
+    }
+
+    :hover {
+        background-color: #f1f4f7;
+    }
+
+    svg {
+        width: 1rem;
+        margin-right: 0.75rem;
+    }
+`
+
+const RoomDropdown = ({ button }: Props) => {
+    const [isOpen, setIsOpen] = useState(false)
+    const ref = useRef<HTMLDivElement>(null)
+
+    useEffect(() => {
+        function handleClickOutside(event: any) {
+            if (ref.current && !ref.current.contains(event.target)) {
+                setIsOpen(false)
+            }
+        }
+
+        // Bind the event listener
+        document.addEventListener('mousedown', handleClickOutside)
+        return () => {
+            // Unbind the event listener on clean up
+            document.removeEventListener('mousedown', handleClickOutside)
+        }
+    }, [ref])
```

Comment:
`ref` is redundant. Does not change. `ref.current` does but `ref.current = el` won't trigger a render. FYI.

Reply:
@bearni95 that's something you may also wanna know.

Reply:
Does this mean I can just remove the dependency array?

Reply:
Here's the difference
- `useEffect(() => {}, [])` runs on mount only
- `useEffect(() => {}, [x])` runs when `x` changed, on update
- `useEffect(() => {})` runs every time, on update

cc @bearni95

Reply:
tl;dr No, leave it empty. :)